src/models/governance-voting.ts

```diff
@@ -0,0 +1,284 @@
+  private createdAtTime?: string;
+
+  static schema = {
+    name: 'GovernanceVoting',
+    properties: {
+      status: 'string',
+      orderNumber: 'int',
+      title: 'string',
+      description: 'string',
+      endDate: 'string',
+      startDate: 'string',
+      votes: 'string?',
+      depositNeeds: 'string?',
+      depositEndTime: 'string?',
+      createdAtTime: 'string?',
+    },
+    primaryKey: 'orderNumber',
+  };
+
+  get timeLeftPercent() {
+    const start =
+      this.dateStart.getTime() > 1000 ? this.dateStart : this.createdAt;
+    const diff = differenceInMilliseconds(
+      this.dateEnd.getTime() > 1000 ? this.dateEnd : this.depositEnd,
+      start,
+    );
+    const timeBeforeEnd = Date.now() - start.getTime();
+    const leftPercent = timeBeforeEnd / diff;
+    return leftPercent * 100;
+  }
+
+  get isDeposited() {
+    return this.status === 'deposited';
+  }
+
+  get isVoting() {
+    return this.status === 'voting';
+  }
+
+  get isVoted() {
+    return false;
+  }
+
+  get depositEnd() {
+    if (this.depositEndTime) {
+      return new Date(this.depositEndTime);
+    } else {
+      return new Date();
+    }
+  }
+
+  get createdAt() {
+    if (this.createdAtTime) {
+      return new Date(this.createdAtTime);
+    } else {
+      return new Date();
+    }
+  }
+
+  get dateEnd() {
+    return new Date(this.endDate);
+  }
+
+  get dateStart() {
+    return new Date(this.startDate);
+  }
+
+  get dataDifference() {
+    const date = new Date(this.endDate);
+
+    const diff = differenceInSeconds(
+      getSeconds(date) > 1 ? date : new Date(this.depositEndTime ?? ''),
+      new Date(Date.now()),
+    );
+
+    const daysLeft = Math.floor(diff / 86400);
+    const hourLeft = Math.floor(diff / (60 * 60)) - daysLeft * 24;
+    const minLeft = Math.floor(diff / 60) - hourLeft * 60 - daysLeft * 24 * 60;
+
+    return {
+      daysLeft,
+      hourLeft,
+      minLeft,
+      isActive: diff > 0,
+    };
+  }
+
+  get yesPercent() {
+    const allVotes = this.proposalVotes;
+    const allVotesSum = Object.values(allVotes).reduce(
+      (acc: number, item: number) => acc + item,
+      0,
+    );
+    return (allVotes.yes / allVotesSum) * 100;
+  }
+
+  get proposalVotes(): VotesType {
+    if (this.votes) {
+      const rawObj = JSON.parse(this.votes);
+      let preparedObj: any = {};
+      Object.keys(rawObj).forEach(key => {
+        preparedObj[key] = isNaN(rawObj[key]) ? 0 : rawObj[key];
+      });
+      return preparedObj;
+    } else {
+      return {
+        yes: 1,
+        no: 1,
+        veto: 1,
+        abstain: 1,
+      };
+    }
+  }
+
+  get proposalDepositNeeds(): number | undefined {
+    if (this.depositNeeds) {
+      return JSON.parse(this.depositNeeds).reduce(
+        (acc: number, item: Coin) => acc + item.amount,
+        0,
+      );
+    } else {
+      return;
+    }
+  }
+
+  async sendDeposit(address: string, amount: number) {
+    const cosmos = new Cosmos(app.provider!);
```

Comment:
IMHO: no need to mix the logic of database models and the logic of obtaining this very data, there are services for this

Reply:
This is done in order not to write an extra parameter to the function and call the method directly from the element without using an extra render

![image](https://user-images.githubusercontent.com/6774813/208142373-8c08ac0c-f6ec-4602-9ddf-f6a11934545c.png)